src/Orchard.Events/DelegateHelper.cs

```diff
@@ -41,7 +41,7 @@ public static class DelegateHelper {
 
         static Func<object, object[], object> BuildFunc<T, TRet>(MethodInfo method) {
 #if DNXCORE50
-            var func = (Func<T, TRet>)method.CreateDelegate(typeof(Func<T, TRet>), method);
+            var func = (Func<T, TRet>)method.CreateDelegate(typeof(Func<T, TRet>));
```

Comment:
```
 static Func<object, object[], object> BuildAction<T, T1, T2>(MethodInfo method) {
#if DNXCORE50
            var func = (Action<T, T1, T2>)method.CreateDelegate(typeof(Action<T, T1, T2>), method);
#else
            var func = (Action<T, T1, T2>)Delegate.CreateDelegate(typeof(Action<T, T1, T2>), method);
#endif
            Func<object, object[], object> ret = (target, p) => { func((T)target, (T1)p[0], (T2)p[1]); return null; };
            return ret;
        }
```
When used DNXCORE50:
```
An exception of type 'System.ArgumentException' occurred in mscorlib.ni.dll but was not handled in user code.
Additional information: Cannot bind to the target method because its signature or security transparency is not compatible with that of the delegate type.
```